### DISCUSSION TITLE
Use getMessageFallback to provide fallback to default locale

### DISCUSSION THREAD

**Discussion by moy2010** [amannn/next-intl#123] [Q&A] [answered]
Hi everyone! I'm wondering if there’s any way to use `getMessageFallback` to provide a fallback to the default locale. 

The reasoning for this is that a key might be missing in one locale file, and I would like to show the English translation as a fallback rather than an error message.

I couldn't find any API that would leverage the fact that we have the namespace and the key for the missing translation.

**Comment by amannn** [comment, marked as answer]
Hello @moy2010 and thanks for the question! As this has come up before, I thought it might be a good idea to include a section about this in the docs.

Can you let me know if this looks useful to you?

https://next-intl-80nf5cxec-amann.vercel.app/docs/usage/error-handling#fallbacks-from-other-locales

Relevant PR: https://github.com/amannn/next-intl/pull/124

**Reply by moy2010** [reply]
Hi Jan! Yes, the PR looks great! Thanks :D

**Reply by moy2010** [reply]
Btw, I came up with a different solution by fetching the english messages on a component wrapping `NextIntlProvider`, so that they could be used within `getMessageFallback`. That way I can still get alerted of a missing key :).

**Reply by amannn** [reply]
Hey, good to hear! That's a viable solution, just note that it might have a performance impact since you're loading two locale bundles now.

Thanks for checking out the PR! 🙂

**Reply by moy2010** [reply]
Indeed. That's the compromise I have to accept for now.
